packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+/// is Flutter's iOS-styled alternative to [ListView].
+///
+/// #Cells
+///
+/// To achieve complete iOS style parity, consider filling [CupertinoTableView]s
+/// with [CupertinoTableViewCell]s. If custom cells are desired, any widget
+/// can be added to a [CupertinoTableView].
+///
+/// #Dividers
+///
+/// By default, [CupertinoTableView]s render dividers between each cell. For
+/// custom cells, the divider beneath the cell is rendered with a standard inset
+/// on the starting edge and is then drawn all the way to the ending edge. For
+/// [CupertinoTableViewCell]s, the divider starts where the title text starts
+/// and then renders all the way to the ending edge. This divider behavior is
+/// as per iOS behavior.
+///
+/// If a [CupertinoTableView] does not contain enough cells to fill the viewport
+/// then extra dividers are rendered beyond the last cell to the end of the
+/// viewport. This extra divider rendering extends into overscroll space at the
+/// end, too. The gap between extra dividers is equal to the height of the
+/// last cell in the [CupertinoTableView].
+///
+/// #Styles
+///
+/// A [CupertinoTableView] may be rendered either in a "plain" style or a
+/// "grouped" style.
+///
+/// A plain [CupertinoTableView] renders a single set of cells without any
+/// divisions between cells.
+///
+/// A grouped [CupertinoTableView] subdivides cells into sections. Sections
+/// are rendered with gaps between them and optional headers before each section.
+///
+/// #Section Index
+///
+/// If a [CupertinoTableView] is in a grouped style, it can optionally display
+/// a section "index". This index is a thin strip that appears on the right side
+/// of the [CupertinoTableView], displays titles for each section in the
+/// [CupertinoTableView], and allows users to tap on a given section title to
+/// automatically scroll to that section.
+///
+/// See also:
+///
+///  * [CupertinoTableViewCell], which renders an iOS-style table view cell that
+///    is intended to appear within a [CupertinoTableView].
+///  * [ListView], which renders a style agnostic list.
+class CupertinoTableView extends StatefulWidget {
+  /// Creates a plain-style [CupertinoTableView] with an explicit list of cells.
+  const CupertinoTableView.plain({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildren = const <Widget>[],
+  }) : plainChildrenBuilder = null,
+       plainChildCount = null;
+
+  /// Creates a plain-style [CupertinoTableView] with a lazily built list of cells.
+  const CupertinoTableView.plainBuilder({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildCount,
+    @required this.plainChildrenBuilder,
+  }) : plainChildren = null;
+
+  /// ScrollController that controls the scroll position of this [CupertinoTableView].
+  final ScrollController scrollController;
+
+  /// The table's background color beneath its cells.
+  final Color backgroundColor;
```

Comment:
Need to be careful with the name since the gap has a different color in grouped style. Arguably, the white is the 'foreground' and the gray in the gap between the cells is the background. Better be less generic with the name.

Reply:
So the normal behavior for a UITableViewCell on iOS is to be drawn without its own background.  Thus the white color you see in a UITableViewCell is actually the UITableView's color showing through.

Doesn't that mean that the section dividers are explicitly drawn on top of the UITableView's background making them a separate artifact and owning their own "color" or "backgroundColor"?

Reply:
@xster I wanted to make sure you didn't miss this question here.

Reply:
Depends on how we compose the combination of table/plain sliver/plain trailing dividers sliver/one group with optional sticky header sliver/boxes inside sliver. 

I didn't the read the other replies yet but if we use a delegate to build boxes with dividers inside either the plain sliver or the one group with optional sticky header sliver, then the white is arguably the sliver's background and the gray is the whole table's background (which shows through different groups through the SliverPadding).